### ISSUE TITLE
FxCop rule CA1709: Identifiers should be cased correctly

### ISSUE THREAD

**Issue by Jm-Fox** [dotnet/roslyn-analyzers#1999]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

[CA1709](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1709-identifiers-should-be-cased-correctly?view=vs-2017)

#### Repro steps

1. Clone https://github.com/Jm-Fox/NetCoreCodeAnalysis
2. Open CodeAnalysis.sln (I used VS 2017.9)
3. In the solution explorer, right click on Solution 'CodeAnalysis' (2 projects) and select Analyze > Run Code Analysis on Solution

#### Expected behavior

A CA1709 issue is expected on the type CodeAnalysis.Ca1722Exception in the NetCore project.

#### Actual behavior

No CA1709 issue is found on the NetCore project.

#### Priority
This is moderately important to me and my employer. Having identifiers cased correctly is important to writing readable and maintainable code.

See also #1948

Edit: used wrong word

**Comment by mavasani** [comment]
This rule has not yet been ported, see https://github.com/dotnet/roslyn-analyzers/issues/445. This is one of the many naming rules, none of which are ported. We have an open PR https://github.com/dotnet/roslyn-analyzers/pull/1111 to port first of these rules, and also add infrastructure for porting other naming rules.

Tagging @jinujoseph and @vatsalyaagrawal - naming rules have been brought up multiple times, I think we should make a call on porting these to our analyzer package rather then wait on introducing these in VS in the core editor layer as spell checkers.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/445.

@jinujoseph @vatsalyaagrawal - another set of customers that have requested the FxCop naming/spell check rules to be ported. I agree with @Jm-Fox here that missing porting of naming rules is currently the **biggest** migration blocker from legacy FxCop to FxCop analyzers. We need to investigate or find out which is the best spell checker library that is available for NetCore/NetStandard to unblock porting these rules.
